monix-execution/shared/src/main/scala/monix/execution/CancelableFuture.scala

```diff
@@ -162,4 +164,10 @@ object CancelableFuture {
     override def cancel(): Unit =
       cancelable.cancel()
   }
+
+  implicit def cancelableFutureMonad(implicit ec: ExecutionContext): Monad[CancelableFuture] = new Monad[CancelableFuture] with StackSafeMonad[CancelableFuture] {
```

Comment:
I'm trying to have consistent naming and just made some changes, so according to PR #397 we'll now have:

- `Task.catsInstances` (implementing `Async[Task]`)
- `Task.catsEffectInstances` (implementing `Effect[Task]`, different from above because this one requires a `Scheduler`)
- `Coeval.catsInstances`
- `Observable.catsInstances`
- `Iterant.catsInstances`

It would be good if this where named `catsInstances` too.

Also, we should probably implement `MonadError[CancelableFuture, Throwable]` as well.